css/src/components/banner.scss

```diff
@@ -0,0 +1,100 @@
+$banner-background: $secondary-background !default;
+$banner-padding: $spacer-5 !default;
+
+$banner-font-size: $font-size-7 !default;
+$banner-color: $text !default;
+
+$banner-border-color: $border-white-high-contrast !default;
+$banner-border-width: $border-width !default;
+
+.banner {
+	@extend %layout-gap;
+
+	display: block;
+	position: relative;
+	outline-color: $text;
+	background-color: $banner-background;
+	color: $banner-color;
+	font-size: $banner-font-size;
+	padding-block: $banner-padding;
+	word-wrap: break-word;
+	word-break: break-word;
+	border-block: $banner-border-width solid $banner-border-color;
+
+	&.is-loading {
+		color: transparent !important;
+
+		> :first-child {
+			margin-inline-start: calc($banner-padding + 0.375em);
+		}
+
+		&::before {
+			@include loader;
+
+			position: absolute !important;
+			inset-block-start: $banner-padding;
+			inset-inline-start: $banner-padding;
+			border-color: transparent transparent $banner-color $banner-color !important;
+		}
+	}
+
+	@each $name, $color-set in $colors {
+		$base: nth($color-set, $color-index-base);
+		$dark: nth($color-set, $color-index-dark);
+		$background: nth($color-set, $color-index-background);
+
+		&.banner-#{$name} {
+			background-color: $background;
+
+			@if $name == 'info' or $name == 'primary' {
+				color: $dark;
+			} @else {
+				color: $base;
+			}
+
+			&.is-loading::before {
+				border-color: transparent transparent $dark $dark !important;
+			}
+
+			/* stylelint-disable max-nesting-depth */
+
+			.button-close {
+				color: inherit;
+
+				&:hover,
+				&.is-hovered {
+					color: $dark;
+				}
+			}
+
+			/* stylelint-enable max-nesting-depth */
+		}
+	}
+
+	.banner-title,
+	a:not(.button) {
+		color: currentColor;
```

Comment:
Could we tweak this rule so that in the high contrast theme, links inside banners can stay bright yellow?

<img width="1041" alt="Banner in high contrast theme with a black background, white text, and a bold white link that reads: Link Text" src="https://user-images.githubusercontent.com/18060369/230142697-32806d60-5bd2-41f7-bef7-a8f1b1cda75d.png">

Feel free to resolve if we don't want/need that.

Reply:
Might be worth one of the dreaded theme-selector rules

Reply:
Updated! :)